crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -575,6 +575,155 @@ pub fn impl_reflect_value(input: TokenStream) -> TokenStream {
     )
 }
 
+#[proc_macro]
+pub fn impl_reflect_struct_and_from_reflect_struct(input: TokenStream) -> TokenStream {
+    let mut iter = input.into_iter().peekable();
+    let (r#constructor, ctor) = (iter.next().unwrap(), iter.next().unwrap());
```

Comment:
Another reason to use `Parse` is that if this fails, we don't get any help from the compiler (which might otherwise say something like `expected keyword: Constructor` or something.